plan/validator.go

```diff
@@ -493,3 +477,87 @@ func isIncorrectName(name string) bool {
 	}
 	return false
 }
+
+func (p *preprocessor) tableUniqueName(schema, table model.CIStr) string {
+	if schema.L != "" && schema.L != p.ctx.GetSessionVars().CurrentDB {
+		return schema.L + "." + table.L
+	}
+	return table.L
+}
+
+func (p *preprocessor) handleTableName(tn *ast.TableName) {
+	if tn.Schema.L == "" {
+		currentDB := p.ctx.GetSessionVars().CurrentDB
+		if currentDB == "" {
+			p.err = errors.Trace(ErrNoDB)
+			return
+		}
+		tn.Schema = model.NewCIStr(currentDB)
+	}
+	if p.inCreateOrDropTable {
+		// The table may not exist in create table or drop table statement.
+		// Skip resolving the table to avoid error.
+		return
+	}
```

Comment:
Why remove `inDeleteTableList` if block?

Reply:
inDeleteTableList only exists in DeleteStmt,
we will skip the children of DeleteStmt now in validator.go.

We will handle this when moving DeleteStmt here.